internal/handlers/handler_reset_password_step2.go

```diff
@@ -31,8 +31,8 @@ func ResetPasswordPost(ctx *middlewares.AutheliaCtx) {
 
 	if err != nil {
 		switch {
-		case strings.Contains(err.Error(), ldapPasswordComplexityCode):
-			ctx.Error(fmt.Errorf("%s", err), ldapPasswordComplexityCode)
+		case utils.IsStringInSlice(err.Error(), ldapPasswordComplexityErrors):
```

Comment:
We may need to think about this more, strings.Contains checks if one string contains another, utils.IsStringInSlice checks if the string is a 100% match for any string in the slice. I'm thinking something like this:

```go
// IsStringInSliceContains checks if a single string is in an array of strings.
func IsStringInSliceContains(a string, list []string) (inSlice bool) {
	for _, b := range list {
		if strings.Contains(a, b) {
			return true
		}
	}

	return false
}
```

Reply:
Yep good catch, not only this but it looks like the cases in the frontend need to be revisited and fixed.

Reply:
Addressed in https://github.com/authelia/authelia/pull/1508/commits/0ec0c5d61d6fc6f2d56e6634f292bce326952603.